src/util/error-extractor/index.js

```diff
@@ -0,0 +1,68 @@
+
+class ErrorDetailsExtractor {
+  constructor(extractObjBuilder) {
+    this.status=extractObjBuilder.status;
+    this.messageDetails = { ...extractObjBuilder.messageDetails }
+  }
+
+  static get Builder() {
+    return ErrorDetailsExtractBuilder;
+  }
+  
+}
+
+
+class ErrorDetailsExtractBuilder {
+
+  constructor() {
+    this.status = 0;
+    this.messageDetails = {};
+  }
+  
+  setStatus(status) {
+    this.status = status;
+    return this;
+  }
+
+  /**
+   * This means we need to set a message from a specific field that we see from the destination's response
+   * 
+   * @param {string} fieldPath -- Path of the field which should be set as "error message"
+   * @returns 
+   */
+  setMessageField(fieldPath) {
+    if (this.messageDetails?.message) {
+      // TODO: Should we throw an error here ?
+      // This check basically ensures that "setMessage" was not already before
+      return this;
+    }
+    this.messageDetails = {
+      field: fieldPath
+    }
+    return this;
+  }
+
+  /**
+   * This means we need to set the message provided
+   * 
+   * @param {string} msg - error message
+   * @returns 
+   */
+  setMessage(msg) {
```

Comment:
why is this check needed ?

Reply:
For any matching error response, we would only need either a field in the error response that doesn't have any dynamic component inside it and also explains the problem in a good manner or we would need to set the message ourselves(hard-coded into the source-code)

To prevent having both values set, I have included a check.